### ISSUE TITLE
Advanced filesystem option: boot flag issue

### ISSUE THREAD

**Issue by orschiro** [Antergos/Cnchi#254]
Hi there,

Many thanks for these great developments of Antergos and Cnchi!

Last time I installed Antergos was back in 2012 and I was totally satisfied with it. Until yesterday when I accidentally broke my entire system due to a wrong option in an rsync backup script...

Thus time for a new, clean installation. I used the advanced option at the point of filesystem partitioning.

I used to use only three LVM partitions:

* Volgroup00/Arch
* Volgroup00/Home
* Volgroup00/Swap

`Volgroup00/Arch` was also the place of my /boot partitioning. However, with Cnchi `.35` it is complaining that I don't have a boot partition set. 

Is it possible to add an option in the Installer to manage the flags, i.e. being able to the boot flag to one of the LVM partitions?

As a workaround I created a new partition `Volgroup00/Boot` but would have preferred to stay with 3 partitions.

Warmly,

Robert

**Comment by karasu** [comment]
Hi,

Thanks for you kind words and your feedback. I'll check this because it should let you go on without a boot partition, but if I recall correctly there are some cases where we enforce a boot partition....

I'll look into it asap. Thanks!

**Comment by orschiro** [comment]
Thanks for looking into it. I am very pleased by the new installation. It was worth it!

**Comment by lots0logs** [comment]
We enforce a boot partition when using LVM because grub required it (at that time). I believe grub's LVM support is much better today so we will have to take another look at this for sure.

**Comment by orschiro** [comment]
When using LVM, if your / partition is on the LVM, it used to be
    necessary to have a separate /bootpartition. In such a
    configuration, the /boot partition is not be a partition of the LVM,
    but rather a partition on the disk before the LVM starts. This is
    because boot loaders could not read files from an LVM. So you could
    never boot into your system on the LVM if it didn’t have a separate
    /boot partition (see this
    <https://help.ubuntu.com/11.10/serverguide/C/advanced-installation.html>
    and this
    <http://www.centos.org/docs/5/html/Deployment_Guide-en-US/s1-lvm-diskdruid-manual.html>
    for details).

    The ability to read files from an LVM was added with GRUB2
    <http://www.gnu.org/software/grub/manual/grub.html#Changes-from-GRUB-Legacy>,
    which means that all recent versions of Ubuntu (Ubuntu 9.10 Karmic
    Koala and later) have it <https://help.ubuntu.com/community/Grub2>.
    So with proper configuration you can have your entire Ubuntu system
    in an LVM without a separate /boot partition. See this page
    <https://wiki.archlinux.org/index.php/GRUB2#LVM> for details about
    how to configure this. (In fact, the only currently supported
    <https://wiki.ubuntu.com/Releases> Ubuntu release that uses the
    original GRUB instead of GRUB2 is Ubuntu 8.04 LTS Server; every
    other non-EoL release—10.04 LTS, 10.10, 11.04, and 11.10—uses GRUB2.)

    If you’re using LVM but not for your Ubuntu system drive, just for a
    storage drive, or for some part of your system (perhaps for /home)
    but not /, then a separate /boot partition is not necessary, even if
    you are using an old (pre-GRUB2) system.

    In summary, a separate /boot partition is largely a matter of
    personal preference for systems that do not use LVM, whereas an
    older system installed on an LVM might need one.

Source: 
http://askubuntu.com/questions/76095/what-is-the-use-of-boot-lvm-based-in-partitioning

On 01/20/2015 10:47 PM, Dustin Falgout wrote:

> We enforce a boot partition when using LVM because grub required it 
> (at that time). I believe grub's LVM support is much better today so 
> we will have to take another look at this for sure.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/Antergos/Cnchi/issues/254#issuecomment-70740714>.
>
​

-- 
Sharing my passion & inspiration for

#sustasty lifestyles #transitions
#opensource #degrowth / Love
#bouldering

Flying is old-school and the
world cannot be saved part-time

Co-Creator of @MindfulMeerkats

www.robert.orzanna.de
twitter.com/@orschiro

**Comment by karasu** [comment]
@lots0logs : So, what do we do? As we are using always the (nearly) latest Grub version, it should be ok to let users with lvm not have a boot partition, shouldn't it?

**Comment by lots0logs** [comment]
Yes but I'm not sure if any adjustments will need to be made to the grub install methods. I would consider this an enhancement to be implemented in a future version.
